yarn-project/end-to-end/src/fixtures/snapshot_manager.ts

```diff
@@ -43,34 +43,80 @@ type SnapshotEntry = {
   snapshotPath: string;
 };
 
-export class SnapshotManager {
-  private snapshotStack: SnapshotEntry[] = [];
+export function createSnapshotManager(testName: string, dataPath?: string) {
+  return dataPath ? new SnapshotManager(testName, dataPath) : new MockSnapshotManager(testName);
+}
+
+export interface ISnapshotManager {
+  snapshot<T>(
+    name: string,
+    apply: (context: SubsystemsContext) => Promise<T>,
+    restore?: (snapshotData: T, context: SubsystemsContext) => Promise<void>,
+  ): Promise<void>;
+
+  setup(): Promise<SubsystemsContext>;
+
+  teardown(): Promise<void>;
+}
+
+/** Snapshot manager that does not perform snapshotting, it just applies transition and restoration functions as it receives them. */
+class MockSnapshotManager implements ISnapshotManager {
   private context?: SubsystemsContext;
-  private livePath: string;
   private logger: DebugLogger;
 
-  constructor(testName: string, private dataPath?: string) {
-    this.livePath = this.dataPath ? join(this.dataPath, 'live', testName) : '';
```

Comment:
Main reason for splitting up the SnapshotManager was errors I was getting when trying to write to an empty `livePath`.